src/basic/macro.h

```diff
@@ -430,4 +430,13 @@ assert_cc(sizeof(dummy_t) == 0);
                 _q && _q > (base) ? &_q[-1] : NULL;      \
         })
 
+/* Iterate through each variadic arg. All must be the same type as 'entry' or must be implicitly
+ * convertable. The iteration variable 'entry' must already be defined. */
+#define VA_ARGS_FOREACH(entry, ...)                                     \
+        _VA_ARGS_FOREACH(entry, UNIQ_T(_entries_, UNIQ), UNIQ_T(_current_, UNIQ), ##__VA_ARGS__)
+#define _VA_ARGS_FOREACH(entry, _entries_, _current_, ...)         \
+        for (typeof(entry) _entries_[] = { __VA_ARGS__ }, *_current_ = _entries_; \
```

Comment:
btw, can we make the compound array we allocate here const somehow? if all arguments are constant this would allow the compiler to put it into a constant section in the resulting ELF file.

(i don't see a way, but maybe you have an idea?)

Reply:
I think it would be hard to do that here, at least not without making the macro quite a bit more complex.

I did have some thoughts about trying to add some macros to help with 2 things: 1) make it easier to create macros that use UNIQ_T() to avoid side effects, and 2) make it easy to create macros that can do constant folding. I'll see if I can get something together and open a separate PR for it.